### PR TITLE
重新测试 gas 消耗

### PR DESCRIPTION
**Pull request by jindada1**

### CODE REVIEW

**Comment by KofClubs** [comment]
@jindada1 Good job!
I shall test them tomorrow.

File: README.md [hunk 1]
```diff
@@ -112,41 +112,17 @@ $ truffle test test/Delegate.js
...
+|22                   |     102750        | 1677889  | 76438   | 
+|100                  |     328940        | 1904121  | 259075  | 
+|222                  |     683187        | 2258431  | 613323  | 
+|400                  |     1201028       | 2776372  | 1131181 | 
+|600                  |     1784292       | 3359732  | 1714457 |
+|800                  |     2369053       | 3944597  | 2299217 |
+|1000                 |     2955295       | 4530958  | 2885484 |
+|1200                 |     3543047       | 5118817  | 3473248 |
+|1400                 |     4132296       | 5708151  | 4062509 |
+|1600                 |     4723041       | 6299001  | 4653244 |
```
Comment (KofClubs) [inline README.md:128]: I don’t need to repeat these tests. 👌
This table is well arranged. Maybe we shall draw a scatter plot based on it in the future to clearly indicate the linear relationship between the fee in gas and the number of contract functions called.

File: test/AccTabSizeChanged.js [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+            }
+
+            await Promise.all(tasks).then((results) => {
+                console.log(results.length);
+            });
+
+            num -= batchSize;
+            console.log(`add ${taskNum} fake data`)
+        }
+    };
```
Comment (KofClubs) [inline test/AccTabSizeChanged.js]: It looks like `addFakedatas_` is no longer needed.

File: test/AccTabSizeChanged.js [hunk 3]
```diff
@@ -0,0 +1,172 @@
+const Operations = artifacts.require("Operations");
+const utils = require("../utils")
+
+const scriptName = utils.scriptName(__filename);
+const testData = utils.testData;
+
+const { uint32_id, string_name, uint32_price } = testData.device;
+
+const bytes_type_user = utils.str2bytes("user");
+const bytes_type_owner= utils.str2bytes("owner");
```
Comment (KofClubs) [inline test/AccTabSizeChanged.js]: `bytes_type_owner` declared but not used.
This line shall be commented out or deleted.

File: truffle-config.js [hunk 4]
```diff
@@ -45,7 +45,7 @@ module.exports = {
     // development: {
     //   host: "127.0.0.1",     // Localhost (default: none)
     //   port: 7545,            // Standard Ethereum port (default: none)
-    //   network_id: "*",       // Any network (default: none)
+    //   network_id: "5777",       // Any network (default: none)
```
Comment (KofClubs) [inline truffle-config.js:48]: 🙋‍♂️ Ask, what is the purpose of revising this line?

File: test/test.js [hunk 5]
```diff
@@ -0,0 +1,44 @@
+const Operations = artifacts.require("Operations");
```
Comment (KofClubs) [inline test/test.js]: I think you'd better rename this file. 😊
This file tests the `addAccTab` contract function, at least the file name should be related to it.

**Review by KofClubs — CHANGES_REQUESTED** [review]
@jindada1 Thank you for your efforts.
Please refer to my suggestions.

File: test/AccTabSizeChanged.js (hunk 2 above)
Comment (jindada1) [reply test/AccTabSizeChanged.js]: yes, It is no longer needed.
I thought that running multiple async functions at the same time would speed up the insert operation.
but I got errors below :

```
ProviderError: 
Could not connect to your Ethereum client.
Please check that your Ethereum client:
    - is running
    - is accepting RPC connections (i.e., "--rpc" option is used in geth)
    - is accessible over the network
    - is properly configured in your Truffle configuration file (truffle-config.js) 
```

May be ? truffle's test network doesn't support so many concurrency ?

File: test/AccTabSizeChanged.js (hunk 3 above)
Comment (jindada1) [reply test/AccTabSizeChanged.js]: sure

File: test/test.js (hunk 5 above)
Comment (jindada1) [reply test/test.js]: I will remove it. 🤫

File: truffle-config.js (hunk 4 above)
Comment (jindada1) [reply truffle-config.js:48]: emmmm~
copy & paste

**Pull request closed without merging**